tools/generate_llvm_coverage.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+BLUE='\033[0;34m'
+NC='\033[0m'
+HAS_LLVMCOV=$(cargo install --list | grep cargo-llvm-cov)
+COVERAGE_OUT=reports/llvm
+
+if [ -z "$HAS_LLVMCOV" ]; then
+    echo "cargo-llvm-cov not found, please install it with 'cargo install cargo-llvm-cov'"
+    exit 1
+fi
+
+mkdir -p $COVERAGE_OUT
+
+# generate coverage using llvm
+# RUST_LOG ensures the trace macros are not flagged by the coverage output
```

Comment:
So the `trace!()` macros were the only issue with code coverage? No other log macros?

Reply:
trace is the log level. this "trips" all levels, and has favorable coverage. I suppose i should remove the 'trace' from the comment to be more accurate